tools/autograd/derivatives.yaml

```diff
@@ -1082,6 +1082,16 @@
   self: where(condition, grad, zeros_like(grad, at::MemoryFormat::Preserve))
   other: where(condition, zeros_like(grad, at::MemoryFormat::Preserve), grad)
 
+- name: maximum(Tensor self, Tensor other) -> Tensor
+  condition: non_differentiable
+  self: where((self != self) | ((other == other) & (self >= other)), grad, zeros_like(grad, at::MemoryFormat::Preserve))
+  other: where((self != self) | ((other == other) & (self >= other)), zeros_like(grad, at::MemoryFormat::Preserve), grad)
```

Comment:
The derivatives for `maximum` and `minimum` are implemented according to the `torch.where` analogy.
The reason for using `self != self` is to check `nan`.
If one of the elements being compared is a NaN, then that element is returned by `maximum` and `minimum`  functions.